internal/util/ctxutil/ctxutil.go

```diff
@@ -61,20 +62,32 @@ func SleepWithJitter(ctx context.Context, d time.Duration, attempts int64) {
 	<-sleepCtx.Done()
 }
 
-// DurationWithJitter returns an exponential backoff duration based on retry with random jitter.
-// The maximum sleep is the cap. The minimum duration is at least 100 milliseconds.
+// DurationWithJitter returns an exponential backoff duration based on attempt with random "full jitter".
+// https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
 //
-// Math/rand is good enough because we don't need the randomness to be cryptographically secure.
-func DurationWithJitter(cap time.Duration, retry int64) time.Duration {
-	const base = time.Millisecond * 100
+// The maximum sleep is the cap. The minimum sleep is at least 3 milliseconds.
+// Provided cap must be larger than minimum sleep, and attempt number must be a positive number.
+func DurationWithJitter(cap time.Duration, attempt int64) time.Duration {
+	const base = 100      // ms
+	const minDuration = 3 // ms
```

Comment:
Simplified both values to integers, as doing calculations on durations creates pretty difficult code to read

Reply:
What is the reason for `capDuration` instead of `cap`? all `base`, `minDuration`, `capDuration` `maxDuration` are same unit, so it's a bit odd to have different naming.

This is honestly just super minor.

Reply:
I wanted to store cap in milliseconds in variable, as it's used multiple times in code and calling `cap.Milliseconds()` inside `math.Min` just makes i more difficult to read, we have also `maxDuration` and `minDuration`, what could we do instead?

Reply:
yeah that's fine :hugs: